### PR TITLE
CON-1262 New C2C Exchange Trace Logs

### PR DESCRIPTION
**Pull request by carlos-castillo-jask**
## Purpose of this pull request

Create a new public doc for a new C2C we are releasing. The image sizes need to be adjusted.

Issue number: CON-1262

## Select the type of change:

- [ ] Minor Changes - Typos, formatting, slight revisions
- [ ] Update Content - Revisions and updating sections
- [x] New Content - New features, sections, pages, tutorials
- [ ] Site and Tools - Updates, maintenance, and new packages for the site, Gatsby, React, etc

### CODE REVIEW

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/microsoft-exchange-trace-logs.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
+---
+id: ms-exchange-trace-logs
+title: Microsoft Exchange Trace Logs
+sidebar_label: Microsoft Exchange Trace Logs
+---
+
```
Comment (JainM6) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/microsoft-exchange-trace-logs.md:6]:
```suggestion
import useBaseUrl from '@docusaurus/useBaseUrl';
```

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/microsoft-exchange-trace-logs.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+2. You will see a menu of options for your Azure Active Directory on the left. Click on the `App registrations` link. <br/> <img src={useBaseUrl('img/send-data/ms-exchange-api-perm-step-2.png')} alt="ms-exchange-api-perm-step-2.png" width="450" />
+3. Click on your Azure Application. <br/> <img src={useBaseUrl('img/send-data/ms-exchange-api-perm-step-3.png')} alt="ms-exchange-api-perm-step-3.png" width="450" />
+4. Click on the `API permissions` link in the left pane of your Azure application and click `Add a permission`. <br/> <img src={useBaseUrl('img/send-data/ms-exchange-api-perm-step-4.png')} alt="ms-exchange-api-perm-step-4.png" width="450" />
+5. A right pane slides out. Click `APIs my organization uses`, search for the text `Office 365 Exchange Online` and click on the result. <br/> <img src={useBaseUrl('img/send-data/ms-exchange-api-perm-step-5.png')} alt="ms-exchange-api-perm-step-5.png" width="450" />
```
Comment (kimsauce) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/microsoft-exchange-trace-logs.md:80]:
```suggestion
5. A right pane slides out. Click `APIs my organization uses`, search for the text `Office 365 Exchange Online`, and click on the result. <br/> <img src={useBaseUrl('img/send-data/ms-exchange-api-perm-step-5.png')} alt="ms-exchange-api-perm-step-5.png" width="450" />
```

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/microsoft-exchange-trace-logs.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+```
+https://login.windows.net/common/oauth2/authorize?response_type=code&resource=https://outlook.office365.com&client_id={{client_id}}&redirect_uri=https://localhost
+```
+
+2. If successful, you will be redirected to a blank page with `code` and `session_state` as URL parameters. Write down the `code` value as this is your Oauth 2.0 authorization code you will use to configure the Sumo Logic C2C. This code is over 700 characters in length. Be careful to not copy the `&session_state=` at the end of the code. See below for an example:
+
+```
+https://localhost/?code={{REDACTED_AUTHORIZATION_CODE}}&session_state=50efb7a9-7678-4186-8b5b-abbb16d7b1ca#
+```
```
Comment (kimsauce) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/microsoft-exchange-trace-logs.md:104]:
```suggestion
1. In a browser where you are already authenticated to your Azure portal at https://portal.azure.com, navigate to the URL below, but replace `{{client_id}}` with your `Application (client) ID` from the earlier steps.
   ```
   https://login.windows.net/common/oauth2/authorize?response_type=code&resource=https://outlook.office365.com&client_id={{client_id}}&redirect_uri=https://localhost
   ```
2. If successful, you will be redirected to a blank page with `code` and `session_state` as URL parameters. Write down the `code` value as this is your Oauth 2.0 authorization code you will use to configure the Sumo Logic C2C. This code is over 700 characters in length. Be careful to not copy the `&session_state=` at the end of the code. See below for an example:
   ```
   https://localhost/?code={{REDACTED_AUTHORIZATION_CODE}}&session_state=50efb7a9-7678-4186-8b5b-abbb16d7b1ca#
   ```
```

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/microsoft-exchange-trace-logs.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+11. Verify your API permissions are correct and granted. <br/> <img src={useBaseUrl('img/send-data/ms-exchange-api-perm-step-11.png')} alt="ms-exchange-api-perm-step-11.png" width="450" />
+
+### Oauth 2.0 Generate Authorization Code
+
+:::note
+This authorization code will expire within 1 hour. Please setup your Sumo source before it expires.
+:::
+
+1. In a browser where you are already authenticated to your Azure portal at https://portal.azure.com, navigate to the URL below, but replace `{{client_id}}` with your `Application (client) ID` from the earlier steps.
+
```
Comment (kimsauce) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/microsoft-exchange-trace-logs.md:95]:
```suggestion
```

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/microsoft-exchange-trace-logs.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+```
+https://login.windows.net/common/oauth2/authorize?response_type=code&resource=https://outlook.office365.com&client_id={{client_id}}&redirect_uri=https://localhost
+```
+
```
Comment (kimsauce) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/microsoft-exchange-trace-logs.md:99]:
```suggestion
```

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/microsoft-exchange-trace-logs.md [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+4. Enter a **Name** for the Source. The description is optional.
+
+    <img src={useBaseUrl('img/send-data/ms-exchange-schema-ui.png')} alt="ms-exchange-schema-ui.png" width="450" />
+
+5. (Optional) For **Source Category**, enter any string to tag the output collected from the Source. Category metadata is stored in a searchable field called `_sourceCategory`.
+
+6. (Optional) **Fields.** Click the **+Add Field** link to define the fields you want to associate, each field needs a name (key) and value.
+
+   * ![green check circle.png](/img/reuse/green-check-circle.png) A green circle with a check mark is shown when the field exists in the Fields table schema.
+   * ![orange exclamation point.png](/img/reuse/orange-exclamation-point.png) An orange triangle with an exclamation point is shown when the field doesn't exist in the Fields table schema. In this case, an option to automatically add the nonexistent fields to the Fields table schema is provided. If a field is sent to Sumo that does not exist in the Fields schema it is ignored, known as dropped.
```
Comment (kimsauce) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/microsoft-exchange-trace-logs.md:129]:
```suggestion
1. In Sumo Logic, select **Manage Data \> Collection \> Collection**. 
2. On the Collectors page, click **Add Source** next to a Hosted Collector.
3. Select **MS Exchange Trace Logs**.<br/> <img src={useBaseUrl('img/send-data/ms-exchange-icon.png')} alt="ms-exchange-icon.png" width="200" />
4. Enter a **Name** for the Source. The description is optional.<br/> <img src={useBaseUrl('img/send-data/ms-exchange-schema-ui.png')} alt="ms-exchange-schema-ui.png" width="450" />
5. (Optional) For **Source Category**, enter any string to tag the output collected from the Source. Category metadata is stored in a searchable field called `_sourceCategory`.
6. (Optional) **Fields.** Click the **+Add Field** link to define the fields you want to associate, each field needs a name (key) and value.
   * ![green check circle.png](/img/reuse/green-check-circle.png) A green circle with a check mark is shown when the field exists in the Fields table schema.
   * ![orange exclamation point.png](/img/reuse/orange-exclamation-point.png) An orange triangle with an exclamation point is shown when the field doesn't exist in the Fields table schema. In this case, an option to automatically add the nonexistent fields to the Fields table schema is provided. If a field is sent to Sumo that does not exist in the Fields schema it is ignored, known as dropped.
```

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/microsoft-exchange-trace-logs.md [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+
+10. **Secret** Enter your client secret generated within your Azure Application.
+
+11.  **Oauth 2.0 Authorization Code** Enter a valid authorization code. They expire within 1 hour of creation. The Microsoft Exchange Trace Logs Source will use this temporary code to automatically generate and rotate your API tokens for accessing the data.
+
+12. **Polling Interval Seconds** We recommend leaving this to its defaults. This value controls how often the Microsoft Exchange Trace Logs Source polls the the Microsoft APIs for new data.
+
+13. **Collection Time Offset** Use this value to offset the collection in the past by some amount of time. Microsoft claims data can be missing from the API within the last 24 hours. Set this to `86400` seconds aka 24 hours to help guarantee more complete results or reduce it if you are looking for more real time, but potentially incomplete data.
+
+14. When you are finished configuring the Source, click **Submit**.
```
Comment (kimsauce) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/microsoft-exchange-trace-logs.md:143]:
```suggestion
8. **Application (client) ID** Enter your client ID from your Azure Application. This should be a Globally Unique Identifier aka GUID.
9. **Directory (tenant) ID** Enter your tenant ID from your Azure Application. This should be a Globally Unique Identifier aka GUID.
10. **Secret** Enter your client secret generated within your Azure Application.
11.  **Oauth 2.0 Authorization Code** Enter a valid authorization code. They expire within 1 hour of creation. The Microsoft Exchange Trace Logs Source will use this temporary code to automatically generate and rotate your API tokens for accessing the data.
12. **Polling Interval Seconds** We recommend leaving this to its defaults. This value controls how often the Microsoft Exchange Trace Logs Source polls the the Microsoft APIs for new data.
13. **Collection Time Offset** Use this value to offset the collection in the past by some amount of time. Microsoft claims data can be missing from the API within the last 24 hours. Set this to `86400` seconds aka 24 hours to help guarantee more complete results or reduce it if you are looking for more real time, but potentially incomplete data.
14. When you are finished configuring the Source, click **Submit**.
```

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/microsoft-exchange-trace-logs.md [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+
+## Error types
+
+When Sumo Logic detects an issue, it is tracked by Health Events. The following table shows the three possible error types, the reason the error would occur, if the Source attempts to retry, and the name of the event log in the Health Event Index.
+
+| Type | Reason | Retries | Retry Behavior | Health Event Name |
+|--|--|--|--|--|
```
Comment (kimsauce) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/microsoft-exchange-trace-logs.md:150]:
```suggestion
|:--|:--|:--|:--|:--|
```

**Review by kimsauce — CHANGES_REQUESTED** [review]
Formatting fixes. Please close the gaps in numbered steps.

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/microsoft-exchange-trace-logs.md [hunk 9]
```diff
@@ -0,0 +1,133 @@
+---
+id: ms-exchange-trace-logs
+title: Microsoft Exchange Trace Logs
+sidebar_label: Microsoft Exchange Trace Logs
+---
+import useBaseUrl from '@docusaurus/useBaseUrl';
+The Microsoft Exchange Trace Logs Source collects email trace logs from the “Office 365 reporting web service” via the “MessageTrace report” under “Exchange reports”. Specific API reference information can be found [here](https://learn.microsoft.com/en-us/previous-versions/office/developer/o365-enterprise-developers/jj984335(v=office.15)).
```
Comment (kimsauce) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/microsoft-exchange-trace-logs.md:7]:
```suggestion

import useBaseUrl from '@docusaurus/useBaseUrl';

The Microsoft Exchange Trace Logs Source collects email trace logs from the “Office 365 reporting web service” via the “MessageTrace report” under “Exchange reports”. Specific API reference information can be found [here](https://learn.microsoft.com/en-us/previous-versions/office/developer/o365-enterprise-developers/jj984335(v=office.15).
```

**Comment by JainM6** [comment]
@kimsauce This file needs corrections and all the mages need to be polished, cropped, and resized. I am working on it and will create a new PR for this.

**Pull request closed without merging**